src/App/component/DashBoardComponent/CodeBaseViews.jsx

```diff
@@ -0,0 +1,191 @@
+    const githubRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'github')
+    if (githubRepo !== undefined) {
+      const query = githubRepo.url.split("github.com/")[1]
+      try {
+        const response = await Axios.get(`http://localhost:9100/pvs-api/github/commits/${query}`,
+          { headers: { "Authorization": `${jwtToken}` } })
+        setCommitListData(response.data)
+      } catch (e) {
+        alert(e.response?.status)
+        console.error(e)
+      }
+    }
+  }
+
+  useEffect(() => {
+    if (Object.keys(currentProject).length !== 0) {
+      getCommitFromGitHub()
+    }
+  }, [currentProject, prop.startMonth, prop.endMonth])
+
+  // Get commits' total additions and deletions
+  useEffect(() => {
+    // Only triger the page rendering once
+    const calculateData = async () => {
+      await Promise.all([
+        getAdditions(),
+        getDeletions()
+      ])
+    }
+    calculateData()
+  }, [commitListData, prop.startMonth, prop.endMonth])
+
+  const getAdditions = () => {
+    const { startMonth, endMonth } = prop
+
+    const chartDataset = { additions: 0 }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.additions += (commitListData.filter(commit => {
+        return moment(commit.committedDate).format("YYYY-MM") === month.format("YYYY-MM")
+      })
+        .reduce(function (additionSum, currentCommit) {
+          return additionSum + currentCommit.additions;
+        }, 0))
+    }
+
+    setAdditions(chartDataset)
+  }
+
+  const getDeletions = () => {
+    const { startMonth, endMonth } = prop
+
+    const chartDataset = { deletions: 0 }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.deletions += (commitListData.filter(commit => {
+        return moment(commit.committedDate).format("YYYY-MM") === month.format("YYYY-MM")
+      })
+        .reduce(function (deletionSum, currentCommit) {
+          return deletionSum + currentCommit.deletions;
+        }, 0))
+    }
+
+    setDeletions(chartDataset)
+  }
+
+  const setAdditions = (chartDataset) => {
+    const job = { id: {}, job: {}, views: {} }
+    job.id = '1'
+    job.job = "Additions"
+    job.views = chartDataset.additions
+    setJobs([job])
+  }
+
+  const setDeletions = (chartDataset) => {
+    const job = { id: {}, job: {}, views: {} }
+    job.id = '2'
+    job.job = "Deletions"
+    job.views = chartDataset.deletions
+    setJobs(prevArray => [...prevArray, job])
+  }
+
+  return (
+    <div>
+      <div>
+        <ul className={classes.totalJobViewsGrid}>
+          {jobs?.map(job => {
+            return (
+              <li className={classes.jobViewsBlock} key={job?.id}>
+                <span className={classes.jobTitle}>{job?.job}</span>
+
+                <div className={classes.jobViewsContainer}>
+                  <span className={classes.jobViews}>{job?.views}</span>
```

Comment:
Since we know the `job` is not a `undefined` before, suggest remove these null check operators.
```suggestion
              <li className={classes.jobViewsBlock} key={job.id}>
                <span className={classes.jobTitle}>{job.job}</span>

                <div className={classes.jobViewsContainer}>
                  <span className={classes.jobViews}>{job.views}</span>
```